Makefile: revert libsepol: use ln --relative to create .so symlinks

This reverts 71393a181d63c9baae5fe8dcaeb9411d1f253998

ln --relative is too recent to be available in all distributions,
especially enterprise-grade distros that can stick around as long as
they are maintained (up to 10 years in some cases?).

For the sake of Buildroot, revert the upstream patch.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN a/src/Makefile b/src/Makefile
--- a/src/Makefile
+++ b/src/Makefile
@@ -77,7 +77,7 @@
 ifeq ($(STATIC),)
 	test -d $(SHLIBDIR) || install -m 755 -d $(SHLIBDIR)
 	install -m 755 $(LIBSO) $(SHLIBDIR)
-	ln -sf --relative $(SHLIBDIR)/$(LIBSO) $(LIBDIR)/$(TARGET)
+	cd $(LIBDIR) && ln -sf ../../`basename $(SHLIBDIR)`/$(LIBSO) $(TARGET)
 endif
 
 relabel:
